internal/ls/autoimports.go

```diff
@@ -1409,27 +1414,78 @@ func forEachExternalModuleToImportFrom(
 	// }
 }
 
+func getIsExcludedPatterns(preferences *lsutil.UserPreferences, useCaseSensitiveFileNames bool) []*regexp2.Regexp {
+	if preferences.AutoImportFileExcludePatterns == nil {
+		return nil
+	}
+	var patterns []*regexp2.Regexp
+	for _, spec := range preferences.AutoImportFileExcludePatterns {
+		// The client is expected to send rooted path specs since we don't know
+		// what directory a relative path is relative to.
```

Comment:
We get these prefs raw now, so I'm not sure these are actually rooted from the client. We might need to pull on some VS Code code to root the paths like VS Code used to when sending prefs (unless that's in this PR and I just haven't gotten there yet).